course/models.py

```diff
@@ -451,7 +452,7 @@ def __str__(self):
     def clean(self):
         super().clean()
         errors = {}
-        RESERVED = ("instances",)
+        RESERVED = tuple(item.value for item in ReservedWordsCourseInstances)
```

Comment:
I think the reserved keywords should be defined in the `class CourseInstance`. You have now moved the definition to the `lib/reserved_words.py`, but I feel that the reserved words are part of the CourseInstance and they should be defined in the class. It is true that the reason that the word `instances` is reserved is actually part of the URL config, but those URLs are closely related to courses.

When we compare this to the constants `ENROLLMENT_AUDIENCE` etc. at the start of `class CourseInstance`, we can see that those constants are also used in the field definitions in the same class. Thus, it would not make sense to define those constants outside the class. The reserved keywords are only used in the URL config outside the class and the `clean` (validation) method in this class. Nevertheless, since the reserved keywords are used in this class and the keyword is really related to this class, I would define the keywords in the class as a constant. Note that you can refer to `CourseInstance.ENROLLMENT_AUDIENCE.INTERNAL_USERS` anywhere after importing `CourseInstance`.

Note that A+ has a custom `Enum` class that is different from the class in the Python standard library. The A+ Enum includes description strings for the values too.

Since there is only one keyword now, the `AutoName` trick is really not needed and it can be removed.